### ISSUE TITLE
[Losses] _divide_no_nan turns an infinite loss into 0.0, so broken trials win Auto search

### ISSUE THREAD

**Issue by jnamora** [Nixtla/neuralforecast#1601]
### What happened + What you expected to happen

What happened:
A diverged trial reported loss = 0.0. Its predictions were ±inf, so the loss
should have been inf. Because 0.0 is the lowest possible value, the Auto
search treated the broken trial as the best one, kept it through every ASHA
rung, and returned it as the winner. No warning, no exception. best_config
stored the diverged configuration.

What I expected to happen:
A diverged trial should report inf, or at least a very large loss, so it
ranks worst in the search. The winner should be one of the trials that
actually converged.

### Versions / Dependencies

```
neuralforecast 3.2.1
torch 2.13.0+cu130
ray 2.57.0
python 3.12.3
Linux
```

### Reproduction script

```python
import torch
from neuralforecast.losses.pytorch import MAE

y = torch.tensor([[[10.0], [20.0]]])
print(MAE()(y=y, y_hat=torch.full_like(y, float("inf"))))

# actual:   tensor(0.)
# expected: tensor(inf)
```

### Issue Severity

High: It blocks me from completing my task.
